src/sentry/integrations/slack/message_builder/image_block_builder.py

```diff
@@ -0,0 +1,81 @@
+from sentry.integrations.slack.message_builder.base.block import BlockSlackMessageBuilder
+from sentry.integrations.slack.message_builder.time_utils import get_approx_start_time
+from sentry.issues.grouptype import PerformanceP95EndpointRegressionGroupType
+from sentry.models.apikey import ApiKey
+from sentry.models.group import Group
+from sentry.snuba.referrer import Referrer
+
+logger = logging.getLogger("sentry.integrations.slack")
+
+
+class ImageBlockBuilder(BlockSlackMessageBuilder):
+    def __init__(self, group: Group) -> None:
+        super().__init__()
+        self.group = group
+        self.event = group.get_recommended_event_for_environments()
+
+    def build(self) -> SlackBlock | None:
+        if (
+            features.has("organizations:slack-endpoint-regression-image", self.group.organization)
+            and self.group.issue_type == PerformanceP95EndpointRegressionGroupType
+        ):
+            return self._build_endpoint_regression_image_block()
+
+        # TODO: Add support for other issue alerts
+        logger.warning("build_issue_alert_image.not_supported")
+        return None
+
+    def _build_endpoint_regression_image_block(self) -> SlackBlock | None:
+        logger.info(
+            "build_endpoint_regression_image.attempt",
+            extra={
+                "group_id": self.group.id,
+            },
+        )
+        endpoint = "events-stats/"
+        organization = self.group.organization
+        try:
+            resp = client.get(
+                auth=ApiKey(organization_id=organization.id, scope_list=["org:read"]),
+                user=None,
+                path=f"/organizations/{organization.slug}/{endpoint}",
+                data={
+                    "yAxis": ["count()", "p95(transaction.duration)"],
+                    "referrer": Referrer.API_ALERTS_CHARTCUTERIE,
+                    "query": f"event.type:transaction transaction:{self.event.transaction}",
+                    "project": self.group.project.id,
+                    "start": get_approx_start_time(self.group),
+                    "end": datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
+                    "dataset": "metrics",
+                },
+            )
+
+            url = charts.generate_chart(
+                ChartType.SLACK_PERFORMANCE_ENDPOINT_REGRESSION,
+                data={
+                    "evidenceData": self.event.occurrence.evidence_data,
+                    "percentileData": resp.data["p95(transaction.duration)"]["data"],
+                },
+            )
+
+            return self.get_image_block(
+                url=url,
+                title=self.group.title,
+                alt="P95(transaction.duration)",
+            )
+
+        except Exception as e:
+            logger.exception(
+                "build_endpoint_regression_image.failed",
+                extra={
+                    "exception": e,
+                },
+            )
```

Comment:
does this send the error to sentry as well? a log is less discoverable compared to an issue in sentry

Reply:
`logger.error` will also emit a sentry error fwiw and create a log with level `error`